src/pipeline.cc

```diff
@@ -686,6 +690,15 @@ class PipelineWorker : public AsyncWorker {
       image = extractedPost;
     }
 
+    // Threshold - must happen before blurring, due to the utility of blurring after thresholding
+    if (shouldThreshold) {
+        VipsImage *thresholded;
+        if (Threshold(hook, image, &thresholded, baton->threshold)) {
+            return Error();
+        }
+        image = thresholded;
```

Comment:
lol -- I did the same thing, twice, when reviewing the code myself after the heap corruption thing.